### PR TITLE
Fixes volume pump runtime. remove_ratio() will return a new gasmixture instead of null if the parameter is <= 0.

### PR DESCRIPTION
**Pull request by Pickle-Coding**
## About The Pull Request
remove_ratio() returns null if the parameter is 0 for some reason (in this case, transfer_rate being 0), which will shortly cause a runtime since something called removed.total_moles() when removed is null. This changes remove_ratio() to return a new gasmixture instead of null when parameter is below or equal to 0.

## Why It's Good For The Game
Less runtimes.

## Changelog

:cl:
fix: Setting volume pump to 0L/S will no longer runtime.
/:cl:

### CODE REVIEW

**Comment by Pickle-Coding** [comment]
Is there a reason why remove_ratio(0) should return null? Would change that instead if I know there is no reason not to.

File: code/modules/atmospherics/machinery/components/binary_devices/volume_pump.dm [hunk 1]
```diff
@@ -73,6 +73,9 @@
 
 	var/transfer_ratio = transfer_rate / air1.volume
 
+	if(!transfer_ratio)
+		return
+
 	var/datum/gas_mixture/removed = air1.remove_ratio(transfer_ratio)
 
 	if(!removed.total_moles())
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/components/binary_devices/volume_pump.dm:81]: Rather then checking if transfer ratio is null, you should check if removed is null

**Comment by LemonInTheDark** [comment]
> Is there a reason why remove_ratio(0) should return null? Would change that instead if I know there is no reason not to.

Not really, should just return a gasmixture with no gas in it

File: code/modules/atmospherics/gasmixtures/gas_mixture.dm [hunk 2]
```diff
@@ -192,7 +192,9 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 ///Returns: gas_mixture with the gases removed
 /datum/gas_mixture/proc/remove_ratio(ratio)
 	if(ratio <= 0)
-		return null
+		var/datum/gas_mixture/removed = new type
+		removed.volume = volume
+		return removed
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/gasmixtures/gas_mixture.dm:197]: Volume is an argument of gasmixtures. Just add (volume) to the new

**Pull request merged**